DOC: Roadmap documentation

arokem opened the pull request:
The idea behind this PR is to discuss a road-map and document it.

arokem commented:
OK - this ready for review

oesteban commented on README.rst:
```diff
@@ -36,15 +36,13 @@ of results.
 `FreeSurfer <https://surfer.nmr.mgh.harvard.edu/>`__,
-`AFNI <https://afni.nimh.nih.gov/>`__,
-and `Nilearn <https://nilearn.github.io/>`__.
+and `DIPY <https://dipy.org/>`__.
```
Should we just not mention any of these here (so long the citations are clearly stated in the boilerplate)?

arokem replied:
Yes. I think that would be fine. Implemented in 5431efc

oesteban commented on README.rst:
```diff
@@ -36,15 +36,13 @@ of results.
-dMRIPrep performs basic preprocessing steps (coregistration, normalization, unwarping,
-segmentation, skullstripping etc.) providing outputs that can be
-easily submitted to a variety of tractography algorithms.
+dMRIPrep performs basic preprocessing steps (head motion correction,
```
```suggestion
dMRIPrep performs basic preprocessing steps (:abbr:`HMC (Head-Motion Correction)`,
```

oesteban commented on README.rst:
```diff
@@ -36,15 +36,13 @@ of results.
-segmentation, skullstripping etc.) providing outputs that can be
-easily submitted to a variety of tractography algorithms.
+dMRIPrep performs basic preprocessing steps (head motion correction,
+susceptibility distortion correction, eddy current correction,  etc.) providing
```
```suggestion
:abbr:`SDC (Susceptibility-derived Distortion Correction)`,
:abbr:`ECC (Eddy-Currents Correction)`,  etc.) providing
```

oesteban commented on docs/roadmap.rst:
```diff
@@ -0,0 +1,72 @@
+At this point, the processing pipeline will include the following major steps:
+
+#. Susceptibility distortion correction.
+    Using SDC flows
```
```suggestion
    Using `SDCFlows <https://github.com/poldracklab/sdcflows>`__
```

oesteban commented on docs/roadmap.rst:
```diff
@@ -0,0 +1,72 @@
+Version 0.4 (April 1st, 2020)
+-----------------------------
+#. Head motion correction.
+
```
```suggestion

    A SHOREline-based approach, ported from QSIPREP.

```

oesteban commented on docs/roadmap.rst:
```diff
@@ -0,0 +1,72 @@
+
+#. Eddy current correction.
+
+    For this release, these two will be done doing something related to the
```
```suggestion
    We will explore the possible adaptations of the HMC based on SHOREline above.
```

oesteban commented on docs/roadmap.rst:
```diff
@@ -0,0 +1,72 @@
+#. Eddy current correction.
+
+    For this release, these two will be done doing something related to the
+    SHOREline approach In cases where the data are "shelled", 3dSHORE will be
```
```suggestion
    In cases where the data are "shelled", 3dSHORE will be
```

oesteban replied:
Perhaps move these three lines up, under HMC?

oesteban commented on docs/roadmap.rst:
```diff
@@ -0,0 +1,72 @@
+    used as the diffusion model. If the data are single-shell, we will use SFM
+    as the diffusion model.
+
+#. Identification of outliers.
```
```suggestion
#. Framewise-displacement calculation
```

I think the definition of outlier is too vague for a roadmap item.

oesteban commented on docs/roadmap.rst:
```diff
@@ -0,0 +1,72 @@
+
+#. Registration between dMRI and T1w image.
+
+#. Identification of outlier measurements (+ imputation?)
```
is this outlier in the sense of head motion? we should be more clear about that

arokem replied:
That, but also other artifacts (e.g., dropped slices, zippers and so on). I've tried to clarify in 41ee304

oesteban commented on docs/roadmap.rst:
```diff
@@ -0,0 +1,72 @@
+
+- Gradient non-linearity correction
+
+- Bias field correction
```
```suggestion
- B1 inhomogeneity field estimation and :abbr:`INU (intensity non-uniformity) correction`
```

oesteban commented on docs/roadmap.rst:
```diff
@@ -0,0 +1,72 @@
+
+#. Susceptibility distortion correction.
+    Using SDC flows
+
```
I would add here:
```
#. Signal drift estimation
    Leveraging the :math:`b=0` extraction, rescaling and averaging that was merged in `#50 <https://github.com/nipreps/dmriprep/pull/50>`__
```

oesteban commented on README.rst:
```diff
@@ -36,15 +36,13 @@ of results.
-easily submitted to a variety of tractography algorithms.
+dMRIPrep performs basic preprocessing steps (head motion correction,
+susceptibility distortion correction, eddy current correction,  etc.) providing
+outputs that can be easily submitted to a variety of tractography algorithms.
```
```suggestion
outputs that can be easily submitted to a variety of diffusion models.
```

oesteban commented on docs/roadmap.rst:
```diff
@@ -0,0 +1,72 @@
+
+Long-term plans
+---------------
+
```
please remove line

oesteban commented on docs/roadmap.rst:
```diff
@@ -0,0 +1,72 @@
+
+Version 1.0 (Targetted for September 2020)
+------------------------------------------
+
```
please remove line

oesteban commented on docs/roadmap.rst:
```diff
@@ -0,0 +1,72 @@
+In the long run we would like to explore the following processing steps:
+
+- Gibbs ringing (using DIPY's image-based implementation).
+
```
Please remove newlines between enumeration items.

oesteban commented on docs/roadmap.rst:
```diff
@@ -0,0 +1,72 @@
+--------------------
+Development road map
+--------------------
+
```
I believe this newline should also go away

arokem commented:
Thanks for the review @oesteban! I believe I've addressed your comments and also rebased on master. Apologies for all the merge commits. I am still not used to working with the "accept suggestion" feature.

arokem commented:
Hmm. I think that we might be running into issues including  certain sphinx directives in the README file, because it's also being used as the `long_description`: https://app.circleci.com/jobs/github/nipreps/dmriprep/650. Should I remove these abbreviations here?

oesteban commented:
> Should I remove these abbreviations here?

Sure!
